R/orderlyweb_run.R

```diff
@@ -33,25 +33,41 @@ report_run_wait <- function(path, name, key, client, timeout, poll, open,
 }
 
 
-report_run_query <- function(ref, update, timeout, instance) {
-  query <- list()
-  if (!is.null(ref)) {
-    query$ref <- ref
-  }
-  if (!update) {
-    query$update <- "false"
-  }
+report_run_query <- function(timeout) {
   if (!is.null(timeout)) {
     assert_scalar_integer(timeout)
-    query$timeout <- as.character(timeout)
+    query <- list(
+      timeout = as.character(timeout)
+    )
+  } else {
+    query <- NULL
+  }
+  query
+}
+
+report_run_body <- function(parameters, ref, instance) {
+  body <- list()
+  if (!is.null(ref)) {
+    assert_scalar_character(ref)
+    body$gitCommit <- ref
   }
   if (!is.null(instance)) {
-    query$instance <- instance
+    ## Instance can be single string
+    ## Or named of instances e.g. list(source = x, annex = f)
+    if (is.character(instance)) {
+      ## TODO: name "source" should not be hard coded and this
+      ## default should depend on the particular orderly configuration
+      ## see VIMC-4587
+      body$instances <- list(source = instance)
```

Comment:
It's not ideal that we are hardcoding `source` name here. @EmmaLRussell do the instances in the body need to have a name? 

I've hardcoded it for now as it would be good to get a fix in but something we should try and remove soon